cabal-install/Distribution/Client/Dependency/Modular/Dependency.hs

```diff
@@ -361,31 +335,29 @@ instance ResetGoal Dep where
 instance ResetGoal Goal where
   resetGoal = const
 
--- | Compute a conflict set from a goal. The conflict set contains the closure
--- of goal reasons as well as the variable of the goal itself.
+-- | Compute a singleton conflict set from a goal, containing just
+-- the goal variable.
 toConflictSet :: Ord qpn => Goal qpn -> ConflictSet qpn
-toConflictSet (Goal g grs) = S.insert (simplifyVar g) (goalReasonChainToVars grs)
+toConflictSet (Goal g _gr) = CS.singleton (simplifyVar g)
```

Comment:
The call to `simplifyVar` is redundant.

Also, I propose to rename this to `goalVarToConflictSet`, hopefully making it more obvious from the name that it only makes use of the variable.